front/src/index.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import './index.css';
+import { RuleBranch, ruleBranchFromRules, RuleLeaf } from './potential_rules';
+import { Setting } from './setting';
+import { App } from './app';
+
+export interface ModelGetSettings {
```

Comment:
This file should only include the app + root element rendering.
I recommend moving all the interfaces and the ruleSet class into separated file/s. pay attention you don't need to use `tsx` extension for pure js functions/classes/